### PR TITLE
Edit and clarify some changes.

### PR DESCRIPTION
**Pull request by gguillotte**
-   Break up and re-order descriptions of numbered and latest collections to make the differences more distinct.
-   Expand and restyle the version pinning example.

### CODE REVIEW

File: source/_includes/puppet-collections/_puppet_collections_intro.md [hunk 1]
```diff
@@ -2,15 +2,31 @@
...
+-   **Numbered collections, such as Puppet Collection 1 (PC1),** are long-lived, stable repositories from which Long-Term Support (LTS) Puppet Enterprise releases are built. Numbered collections maintain the same major version of each component package during its lifetime, which delivers bug fixes and minimizes breaking changes, but also introduces fewer new features.
+-   **The "latest" collection** follows every release of Puppet Enterprise, including versions not considered LTS releases, and is updated with new major-version releases that might introduce breaking changes.
+
+Puppet publishes updates for operating systems starting from the time a package is first published for that operating system to a collection repository, and stops updating those packages 30 days after the end of the operating system's vendor-determined lifespan.
```
Comment (gguillotte) [inline source/_includes/puppet-collections/_puppet_collections_intro.md:12]: Is this true for both numbered and latest collections, or latest collections only?

File: source/_includes/puppet-collections/_puppet_collections_intro.md [hunk 2]
```diff
@@ -2,15 +2,31 @@
...
+
+See [The Puppet Enterprise Lifecycle](https://puppet.com/misc/puppet-enterprise-lifecycle) for information about PE release trains.
```
Comment (gguillotte) [inline source/_includes/puppet-collections/_puppet_collections_intro.md:14]: I'm not sure about the use of "release trains" here, but I don't know what the new lifecycle document looks like.

File: source/_includes/puppet-collections/_puppet_collections_intro.md (hunk 1 above)
Comment (ahpook) [reply source/_includes/puppet-collections/_puppet_collections_intro.md:12]: both, our support is tied to the upstream support.

File: source/_includes/puppet-collections/_puppet_collections_intro.md (hunk 2 above)
Comment (ahpook) [reply source/_includes/puppet-collections/_puppet_collections_intro.md:14]: maybe "release cycle"? we call them trains internally and i believe it'll be the same for the published docs as well.

**Comment by ahpook** [comment]
Thanks @gguillotte  - this is great, I'll merge it and push to the upstream PR.

**Pull request merged**